File: src/01-easy/ascii-art/solution.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;
int main(){
    int L,H;cin>>L>>H;cin.ignore();
    string T;getline(cin,T);
    string rows[30];
    for(int i=0;i<H;i++)getline(cin,rows[i]);
    for(int i=0;i<H;i++){
        for(char c:T){
            int idx=toupper(c)-'A';if(idx<0||idx>25)idx=26;
            cout<<rows[i].substr(idx*L,L);
        }
        cout<<endl;
    }
}

File: src/01-easy/ascii-art/solution.hs
import Data.Char(toUpper,ord)
main = do
  l <- readLn :: IO Int
  h <- readLn :: IO Int
  t <- fmap (map toUpper) getLine
  rows <- sequence [getLine | _ <- [1..h]]
  mapM_ (\row -> putStrLn $ concatMap (\c ->
    let idx = ord c - 65; i = if idx<0||idx>25 then 26 else idx
    in take l (drop (i*l) row)) t) rows
